File: examples/zig/state-ripemd160/HashRegistry.runar.zig
const runar = @import("runar");

pub const HashRegistry = struct {
    pub const Contract = runar.StatefulSmartContract;

    currentHash: runar.Ripemd160 = "",

    pub fn init(currentHash: runar.Ripemd160) HashRegistry {
        return .{ .currentHash = currentHash };
    }

    pub fn update(self: *HashRegistry, newHash: runar.Ripemd160) void {
        self.currentHash = newHash;
        runar.assert(true);
    }
};

File: examples/rust/state-ripemd160/HashRegistry.runar.rs
use runar::prelude::*;

#[runar::contract]
struct HashRegistry {
    current_hash: Ripemd160,
}

impl HashRegistry {
    pub fn update(&mut self, new_hash: Ripemd160) {
        self.current_hash = new_hash;
        assert!(true);
    }
}
